share/lib/python/neuron/__init__.py

```diff
@@ -102,6 +102,15 @@
 import sys
 import os
 
+if sys.version_info[0] == 2:
+    import warnings
+
+    warnings.simplefilter("always", DeprecationWarning)
+    warnings.warn(
+        "Python2 support will be dropped soon! We strongly advise switching to Python3 as of now.",
+        DeprecationWarning,
+    )
+
```

Comment:
Discussed offline : should be moved to autotools and cmake as a build time warning.